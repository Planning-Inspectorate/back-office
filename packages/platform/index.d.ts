import './config/dotenv';
import './http/ttl';
import './util/CryptoUtils';
import './util/date';
import './util/filter';
import './util/url';
import './validators/date';
import './validators/postcode';

declare module '@pins/platform' {
	export * from './config/dotenv';
	export * from './http/ttl';
	export * from './util/CryptoUtils';
	export * from './util/date';
	export * from './util/filter';
	export * from './util/url';
	export * from './validators/date';
	export * from './validators/postcode';
}
