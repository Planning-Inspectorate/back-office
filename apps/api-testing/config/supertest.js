// @ts-nocheck
import supertest from 'supertest';
import config from './config.cjs';
import { Logger } from '../utils/logger.js';

const logger = new Logger();
logger.log('green', '************* Tests starting ****************');
logger.log('cyan', `Running tests against: ${config.baseUrl}`);
logger.log('green', '*********************************************');

const request = supertest(config.baseUrl);

export default request;
