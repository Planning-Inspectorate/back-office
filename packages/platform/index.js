export * from './http/hooks.js';
export * from './testing/assets/assets.js';
export * from './testing/fake.js';
export * from './testing/html-parser.js';
export * from './utils/date.js';
export * from './utils/dotenv.js';
export * from './utils/filter.js';
export * from './validators/date.js';
export * from './validators/postcode.js';
