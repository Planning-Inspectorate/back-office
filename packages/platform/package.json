{
  "name": "@pins/platform",
  "version": "0.0.0",
  "type": "module",
  "main": "index.js",
  "private": true,
  "engines": {
    "node": ">=16.14.0"
  },
  "scripts": {
    "clean": "rimraf .turbo node_modules dist",
    "lint:js": "npx eslint .",
    "lint:js:fix": "npx eslint . --fix",
    "tscheck": "npx tsc -p jsconfig.json --maxNodeModuleJsDepth 0"
  },
  "dependencies": {
    "date-fns": "2.28.0",
    "dotenv": "16.0.0",
    "got": "^12.5.2",
    "html": "^1.0.0",
    "lodash-es": "^4.17.21",
    "node-html-parser": "^5.3.3",
    "randexp": "^0.5.3"
  },
  "devDependencies": {
    "@types/html": "^1.0.1",
    "@types/lodash-es": "^4.17.6",
    "type-fest": "^2.12.2"
  }
}
