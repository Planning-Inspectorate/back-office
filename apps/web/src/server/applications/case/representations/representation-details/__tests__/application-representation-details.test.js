import { parseHtml } from '@pins/platform';
import nock from 'nock';
import supertest from 'supertest';
import { createTestEnvironment } from '../../../../../../../testing/index.js';

const { app, installMockApi, teardown } = createTestEnvironment();
const request = supertest(app);

// Arrange
const mockAPIResponse = {
	id: 1,
	reference: 'BC0110001-3',
	status: 'VALID',
	redacted: true,
	received: '2023-04-14T15:52:56.507Z',
	originalRepresentation:
		'The proposals size, loss of food producing land, battery storage size and technology, joined up thinking between DCO applications',
	redactedRepresentation:
		'The proposals size, loss of food producing land, battery storage size and technology, joined up thinking between DCO applications',
	user: {
		azureReference: 1
	},
	contacts: [
		{
			type: 'PERSON',
			firstName: 'Arthur',
			lastName: 'Test',
			organisationName: null,
			jobTitle: null,
			under18: false,
			email: 'test@example.com',
			phoneNumber: '01234 567890',
			address: {
				addressLine1: '21 The Pavement',
				addressLine2: null,
				town: null,
				county: 'Wandsworth',
				postcode: 'SW4 0HY'
			}
		},
		{
			type: 'AGENT',
			firstName: 'James',
			lastName: 'Bond',
			organisationName: '',
			jobTitle: null,
			under18: false,
			email: 'test-agent@example.com',
			phoneNumber: '01234 567890',
			address: {
				addressLine1: '96 The Avenue',
				addressLine2: 'Maidstone',
				town: null,
				county: 'Kent',
				postcode: 'MD21 5XY'
			}
		}
	],
	attachments: [],
	redactedBy: {
		azureReference: 1
	}
};
// import { representationsFixture } from '../../__fixtures__/representations.fixture.js';
const mockCaseReference = { title: 'mock title', status: 'in test', reference: 'mock reference' };
const nocks = () => {
	nock('http://test/').get('/applications/1').reply(200, mockCaseReference);
	nock('http://test/')
		.get(`/applications/1/representations/1`)
		.reply(200, mockAPIResponse)
		.persist();
};

describe('Representation details page', () => {
	beforeEach(installMockApi);
	afterEach(teardown);

	afterAll(() => {
		nock.cleanAll();
	});

	const baseUrl = '/applications-service/case/1/relevant-representations/1/representation-details';

	describe('GET /applications-service/case/1/relevant-representations/1/representations-details', () => {
		beforeEach(async () => {
			nocks();
			await request.get('/applications-service/case-team');
		});

		it('should render the page', async () => {
			const response = await request.get(baseUrl);
			const element = parseHtml(response.text);

			expect(element.innerHTML).toMatchSnapshot();
		});
	});
});
