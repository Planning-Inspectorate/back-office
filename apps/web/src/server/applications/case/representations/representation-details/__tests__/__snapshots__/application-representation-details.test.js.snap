// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Representation details page GET /applications-service/case/1/relevant-representations/1/representations-details should render the page 1`] = `
"<main class=\\"govuk-main-wrapper \\" id=\\"main-content\\" role=\\"main\\">
    <div class=\\"govuk-grid-row govuk-!-margin-bottom-6\\">
        <div class=\\"govuk-grid-column-full\\">
            <h1 class=\\"govuk-heading-l govuk-!-margin-bottom-2\\"> Mrs Sue</h1><strong class=\\"govuk-heading-s govuk-!-margin-bottom-6\\"><strong class=\\"govuk-tag govuk-tag--grey\\">awaiting review</strong></strong>
            <p             class=\\"govuk-!-margin-bottom-6\\"><strong>Project name:</strong> mock title</p>
        </div>
    </div>
    <div class=\\"govuk-grid-row govuk-!-margin-bottom-6\\">
        <div class=\\"govuk-grid-column-full\\">
            <h3 class=\\"govuk-heading-m\\">Contact details</h3>
            <dl class=\\"govuk-summary-list\\">
                <div class=\\"govuk-summary-list__row govuk-summary-list__row--no-border\\"><dt class=\\"govuk-summary-list__key\\"> Organisation name</dt>
                    <dd class=\\"govuk-summary-list__value\\"></dd>
                    <dd class=\\"govuk-summary-list__actions\\"><a class=\\"govuk-link\\" href=\\"#\\"> Change<span class=\\"govuk-visually-hidden\\"> organisation name</span></a>
                    </dd>
                </div>
                <div class=\\"govuk-summary-list__row govuk-summary-list__row--no-actions govuk-summary-list__row--no-border\\"><dt class=\\"govuk-summary-list__key\\"> Name</dt>
                    <dd class=\\"govuk-summary-list__value\\">Mrs Sue</dd>
                </div>
                <div class=\\"govuk-summary-list__row govuk-summary-list__row--no-actions govuk-summary-list__row--no-border\\"><dt class=\\"govuk-summary-list__key\\"> Job title</dt>
                    <dd class=\\"govuk-summary-list__value\\"></dd>
                </div>
                <div class=\\"govuk-summary-list__row govuk-summary-list__row--no-actions govuk-summary-list__row--no-border\\"><dt class=\\"govuk-summary-list__key\\"> Email address</dt>
                    <dd class=\\"govuk-summary-list__value\\">test@example.com</dd>
                </div>
                <div class=\\"govuk-summary-list__row govuk-summary-list__row--no-actions\\"><dt class=\\"govuk-summary-list__key\\"> Telephone number</dt>
                    <dd class=\\"govuk-summary-list__value\\">01234 567890</dd>
                </div>
                <div class=\\"govuk-summary-list__row\\"><dt class=\\"govuk-summary-list__key\\"> Address</dt>
                    <dd class=\\"govuk-summary-list__value\\">44 Rivervale Bridport DT6 5RN</dd>
                    <dd class=\\"govuk-summary-list__actions\\"><a class=\\"govuk-link\\" href=\\"#\\"> Change<span class=\\"govuk-visually-hidden\\"> address</span></a>
                    </dd>
                </div>
                <div class=\\"govuk-summary-list__row\\"><dt class=\\"govuk-summary-list__key\\"> Preferred contact method</dt>
                    <dd                     class=\\"govuk-summary-list__value\\">Email</dd>
                        <dd class=\\"govuk-summary-list__actions\\"><a class=\\"govuk-link\\" href=\\"edit/preferred-contact\\"> Change<span class=\\"govuk-visually-hidden\\"> preferred preferred contact method</span></a>
                        </dd>
                </div>
                <div class=\\"govuk-summary-list__row\\"><dt class=\\"govuk-summary-list__key\\"> Type</dt>
                    <dd class=\\"govuk-summary-list__value\\"></dd>
                    <dd class=\\"govuk-summary-list__actions\\"><a class=\\"govuk-link\\" href=\\"edit/preferred-contact\\"> Change<span class=\\"govuk-visually-hidden\\"> type</span></a>
                    </dd>
                </div>
                <div class=\\"govuk-summary-list__row\\"><dt class=\\"govuk-summary-list__key\\"> Representation on behalf of</dt>
                    <dd                     class=\\"govuk-summary-list__value\\"></dd>
                        <dd class=\\"govuk-summary-list__actions\\"><a class=\\"govuk-link\\" href=\\"edit/preferred-contact\\"> Change<span class=\\"govuk-visually-hidden\\"> representation on behalf of</span></a>
                        </dd>
                </div>
                <div class=\\"govuk-summary-list__row\\"><dt class=\\"govuk-summary-list__key\\"> Under 18</dt>
                    <dd class=\\"govuk-summary-list__value\\">No</dd>
                    <dd class=\\"govuk-summary-list__actions\\"><a class=\\"govuk-link\\" href=\\"edit/preferred-contact\\"> Change<span class=\\"govuk-visually-hidden\\"> under 18</span></a>
                    </dd>
                </div>
            </dl>
        </div>
    </div>
    <div class=\\"govuk-grid-row govuk-!-margin-bottom-6\\">
        <div class=\\"govuk-grid-column-full\\">
            <h3 class=\\"govuk-heading-m\\">Agent contact details</h3>
            <dl class=\\"govuk-summary-list\\">
                <div class=\\"govuk-summary-list__row govuk-summary-list__row--no-border\\"><dt class=\\"govuk-summary-list__key\\"> Organisation name</dt>
                    <dd class=\\"govuk-summary-list__value\\"></dd>
                    <dd class=\\"govuk-summary-list__actions\\"><a class=\\"govuk-link\\" href=\\"#\\"> Change<span class=\\"govuk-visually-hidden\\">organisation name</span></a>
                    </dd>
                </div>
                <div class=\\"govuk-summary-list__row govuk-summary-list__row--no-actions govuk-summary-list__row--no-border\\"><dt class=\\"govuk-summary-list__key\\"> Name</dt>
                    <dd class=\\"govuk-summary-list__value\\">James Bond</dd>
                </div>
                <div class=\\"govuk-summary-list__row govuk-summary-list__row--no-actions govuk-summary-list__row--no-border\\"><dt class=\\"govuk-summary-list__key\\"> Email address</dt>
                    <dd class=\\"govuk-summary-list__value\\">test-agent@example.com</dd>
                </div>
                <div class=\\"govuk-summary-list__row govuk-summary-list__row--no-actions\\"><dt class=\\"govuk-summary-list__key\\"> Telephone number</dt>
                    <dd class=\\"govuk-summary-list__value\\">01234 567890</dd>
                </div>
                <div class=\\"govuk-summary-list__row\\"><dt class=\\"govuk-summary-list__key\\"> Address</dt>
                    <dd class=\\"govuk-summary-list__value\\">8 The Chase Findon BN14 0TT</dd>
                    <dd class=\\"govuk-summary-list__actions\\"><a class=\\"govuk-link\\" href=\\"#\\"> Change<span class=\\"govuk-visually-hidden\\"> address</span></a>
                    </dd>
                </div>
                <div class=\\"govuk-summary-list__row\\"><dt class=\\"govuk-summary-list__key\\"> Preferred contact method</dt>
                    <dd                     class=\\"govuk-summary-list__value\\">Email</dd>
                        <dd class=\\"govuk-summary-list__actions\\"><a class=\\"govuk-link\\" href=\\"#\\"> Change<span class=\\"govuk-visually-hidden\\"> preferred contact method</span></a>
                        </dd>
                </div>
                <div class=\\"govuk-summary-list__row\\"><dt class=\\"govuk-summary-list__key\\"> Type</dt>
                    <dd class=\\"govuk-summary-list__value\\"></dd>
                    <dd class=\\"govuk-summary-list__actions\\"><a class=\\"govuk-link\\" href=\\"#\\"> Change<span class=\\"govuk-visually-hidden\\"> type</span></a>
                    </dd>
                </div>
                <div class=\\"govuk-summary-list__row\\"><dt class=\\"govuk-summary-list__key\\"> Under 18</dt>
                    <dd class=\\"govuk-summary-list__value\\">No</dd>
                    <dd class=\\"govuk-summary-list__actions\\"><a class=\\"govuk-link\\" href=\\"#\\"> Change<span class=\\"govuk-visually-hidden\\"> under 18</span></a>
                    </dd>
                </div>
            </dl>
        </div>
    </div>
    <div class=\\"govuk-grid-row govuk-!-margin-bottom-6\\">
        <div class=\\"govuk-grid-column-full\\">
            <h3 class=\\"govuk-heading-m\\">Representation</h3>
            <dl class=\\"govuk-summary-list\\">
                <div class=\\"govuk-summary-list__row govuk-summary-list__row--no-actions\\"><dt class=\\"govuk-summary-list__key\\"> Reference</dt>
                    <dd class=\\"govuk-summary-list__value\\">BC0110001-1</dd>
                </div>
                <div class=\\"govuk-summary-list__row govuk-summary-list__row--no-actions\\"><dt class=\\"govuk-summary-list__key\\"> Date received</dt>
                    <dd class=\\"govuk-summary-list__value\\">27 Apr 2023</dd>
                </div>
                <div class=\\"govuk-summary-list__row\\"><dt class=\\"govuk-summary-list__key\\"> Original representation</dt>
                    <dd class=\\"govuk-summary-list__value\\">
                        <div id=\\"representationText\\" class=\\"govuk-!-margin-bottom-2\\">
                            <noscript>Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Aenean commodo
                                ligula eget dolor. Aenean massa. Cum sociis natoque penatibus et magnis
                                dis parturient montes, nascetur ridiculus mus. Donec quam felis, ultricies
                                nec, pellentesque eu, pretium quis, sem. Nulla con.</noscript>
                        </div>
                        <p class=\\"govuk-body govuk-!-margin-bottom-4\\"><a href=\\"#\\" id=\\"fullRepToggle\\" hidden class=\\"govuk-link govuk-link--no-visited-state\\">Show full representation</a>
                        </p>
                        <script nonce=\\"EdcOUaJ8lczj9tIPO0lPow==\\">
                            const repTextEl = document.querySelector('#representationText')
                                                const fullRepToggleEl = document.querySelector('#fullRepToggle')
                            
                                                const repExcerpt = 'Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Aenean commodo ligula eget dolor. Aenean massa. Cum sociis natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus. Donec...'
                                                const repFullText = 'Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Aenean commodo ligula eget dolor. Aenean massa. Cum sociis natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus. Donec quam felis, ultricies nec, pellentesque eu, pretium quis, sem. Nulla con.'
                            
                                                console.log(repExcerpt)
                            
                                                let isToggled = false
                                                fullRepToggleEl.removeAttribute('hidden')
                                                repTextEl.innerText = repExcerpt
                            
                                                
                                                const handleClick =function(e) {
                                                    e.preventDefault() 
                            
                                                    if( !isToggled ) {
                                                        repTextEl.innerText = repFullText
                                                        fullRepToggleEl.innerText = 'Hide full representation'
                                                        isToggled = true
                                                    } else {
                                                        repTextEl.innerText = repExcerpt
                                                        fullRepToggleEl.innerText = 'Show full representation'
                                                        isToggled = false
                                                    }
                                                }
                            
                                                fullRepToggleEl.addEventListener( 'click', handleClick )
                        </script>
                    </dd>
                    <dd class=\\"govuk-summary-list__actions\\"><a class=\\"govuk-link\\" href=\\"edit/preferred-contact\\"> Redact<span class=\\"govuk-visually-hidden\\"> original representation</span></a>
                    </dd>
                </div>
            </dl>
        </div>
    </div>
    <div class=\\"govuk-grid-row govuk-!-margin-bottom-6\\">
        <div class=\\"govuk-grid-column-full\\">
            <h3 class=\\"govuk-heading-m\\">Workflow</h3>
            <dl class=\\"govuk-summary-list\\">
                <div class=\\"govuk-summary-list__row\\"><dt class=\\"govuk-summary-list__key\\"> Status</dt>
                    <dd class=\\"govuk-summary-list__value\\"><strong class=\\"govuk-heading-s govuk-!-margin-bottom-6\\"><strong class=\\"govuk-tag govuk-tag--grey\\">awaiting review</strong></strong>
                    </dd>
                    <dd class=\\"govuk-summary-list__actions\\"><a class=\\"govuk-link\\" href=\\"edit/preferred-contact\\"> Change<span class=\\"govuk-visually-hidden\\"> status</span></a>
                    </dd>
                </div>
            </dl>
            <div><a class=\\"govuk-button govuk-button--secondary\\" href=\\"#\\">View task log</a>
            </div>
        </div>
    </div>
    <div class=\\"govuk-grid-row govuk-!-margin-bottom-6\\">
        <div class=\\"govuk-grid-column-full\\">
            <h3 class=\\"govuk-heading-m\\">Attachments</h3>
            <div><a class=\\"govuk-button govuk-button--secondary\\" href=\\"#\\">Upload files</a>
            </div>
        </div>
    </div>
</main>"
`;
